litgpt/pretrain.py

```diff
@@ -147,71 +193,106 @@ def main(
     out_dir: Path,
     tokenizer_dir: Optional[Path],
     tokenizer: Optional[Tokenizer],
+    use_pt_profiler: bool,
+    pt_profiler_wait: int,
+    pt_profiler_warmup: int,
+    pt_profiler_active: int,
+    pt_profiler_repeat: int,
+    nsys_profile_step_multiple: int,
     train: TrainArgs,
     eval: EvalArgs,
 ) -> None:
-    validate_args(train, eval, initial_checkpoint_dir, resume)
-
-    if fabric.global_rank == 0:
-        out_dir.mkdir(parents=True, exist_ok=True)
+    if use_pt_profiler:
+        cm = nullcontext()
+    else:
+        cm = torch.autograd.profiler.emit_nvtx()
+    with cm:
+        validate_args(train, eval, initial_checkpoint_dir, resume)
+
+        if fabric.global_rank == 0:
+            out_dir.mkdir(parents=True, exist_ok=True)
+        tprofiler_out_dir = out_dir / "tprofiler"
+        execution_trace_out_dir = out_dir / "execution_trace"
+        execution_trace_out_dir.mkdir(parents=True, exist_ok=True)
+
+        fabric.seed_everything(seed)  # same seed for every process to init model (FSDP)
+
+        if use_pt_profiler:
+            prof = tprofiler.profile(
+                schedule=tprofiler.schedule(
+                    wait=pt_profiler_wait,
+                    warmup=pt_profiler_warmup,
+                    active=pt_profiler_active,
+                    repeat=pt_profiler_repeat,
+                ),
+                on_trace_ready=tprofiler.tensorboard_trace_handler(tprofiler_out_dir),
+                record_shapes=True,
+                with_stack=True,
+                activities=[tprofiler.ProfilerActivity.CPU, tprofiler.ProfilerActivity.CUDA],
+                execution_trace_observer=(
```

Comment:
Delete the execution_trace_observer argument

Reply:
Done.